acra-webconfig description (#78)
cossacklabs/product-docs: acra/configuring-maintaining/general-configuration/acra-webconfig.md

---
title: acra-webconfig
bookCollapseSection: true
---

# acra-webconfig

`acra-webconfig` is simple web application that provide web UI for AcraServer's runtime configuration. Currently, it is
under development and for now allows configuring 9 parameters of AcraServer:
* `db_host`
* `db_port`
* `incoming_connection_api_port`
* `d`
* `poison_run_script_file`
* `poison_shutdown_enable`
* `zonemode_enable`

{{< hint info >}}
Parameter's description you can find on AcraServer's [command line flags]({{< ref "/acra/configuring-maintaining/general-configuration/acra_server.md#command-line-flags" >}}) page.
{{< /hint >}}

## Command line flags

* `--destination_host`
  IP or domain to AcraServer/AcraConnector daemon
  Default is `localhost`

> * `--destination_port`
>   Port of AcraServer's HTTP API port or AcraConnector port that proxy API requests to AcraServer
>   Default is `9191`

Reviewer: - Same things as above.
- Port port port port port.

```suggestion
* `--destination_port=<port>`

  HTTP API port of AcraServer or AcraConnector API proxy.
  Default is `9191`.
```

> * `--incoming_connection_host`
> 
>   Host for AcraWebconfig HTTP endpoint
>   Default is `127.0.0.1`

Reviewer: - Typically you pass IP address here since it's unusual to have different hostnames for different interfaces.
  - But as usual, Go thinks different™ and does something unexpected, allowing to specify hostnames here then starts listening on whatever they resolve into.
- It's nice to specify the way to listen on all interfaces as that's a common use case.

```suggestion
* `--incoming_connection_host=<address>`

  Address of the network interface for `acra-webconfig` to listen for HTTP connections on.
  Default is `127.0.0.1`.
  Set to empty string, `0.0.0.0`, or `[::]` to listen on all interfaces.
```

> * `--incoming_connection_port`
> 
>   Port for AcraWebconfig HTTP endpoint
>   Default is `8000`

Reviewer: ```suggestion
* `--incoming_connection_port=<port>`

  Port for `acra-webconfig` to listen for HTTP connections on.
  Default is `8000`.
```

It's also possible to set this to `0` to let the OS choose a port for you, but this port is not logged anywhere, so you can only find it elsewhere. I don't think this feature should be documented then.

> * `--http_auth_mode`
> 
>   Mode for basic auth. Possible values:
>     * `auth_on` (default): turn on basic auth
>     * `auth_off_local`: turn off basic auth if AcraServer/AcraConnector running on same host and accessible via `localhost` or `127.0.0.1`
>     * `auth_off`: turn off basic auth

Reviewer: - Formatting, yada-yada.
- Factual correctness for `auth_off_local`.

```suggestion
* `--http_auth_mode={auth_on|auth_off_local|auth_off}`

  HTTP basic authentication mode.

    * `auth_on` — (default) basic authentication is always required
    * `auth_off_local` — turn off basic authentication if `acra-webconfig` accepts connections only from `localhost`
    * `auth_off` — basic authentication is not performed
```

> * `--static_path`
> 
>   Path to static content
>   Default is `cmd/acra-webconfig/static`

Reviewer: ```suggestion
* `--static_path=<path>`

  Path to static content.
  Default is `cmd/acra-webconfig/static`.
```

### Configuration files

* `--config_file=<filename>`

  Path to YAML configuration file.

> * `--dump_config`
> 
>   Dump configuration to `configs/acra-server.yaml`.

Reviewer: ```suggestion
* `--dump_config`

  Dump configuration to `configs/acra-webconfig.yaml`.
```

> * `--generate_markdown_args_table`
> 
>   Generate markdown file with text description of all flags.
>   Output file is `configs/markdown_acra-server.md`.
>   Works in pair with `--dump_config`.

Reviewer: ```suggestion
* `--generate_markdown_args_table`

  Generate markdown file with text description of all flags.
  Output file is `configs/markdown_acra-webconfig.md`.
  Works in pair with `--dump_config`.
```

### Logging

> * `-d`
> 
>   Log to stderr all `DEBUG`, `INFO`, `WARNING` and `ERROR` logs.

Reviewer: - Based on the source code, it's more like this?

```suggestion
* `-d`

  Log to stderr only `DEBUG` logs and above.
  Do not produce `VERBOSE` logs.
```

Author: actually not. it turns on logs with debug and higher levels - https://github.com/sirupsen/logrus/blob/bdc0db8ead3853c56b7cd1ac2ba4e11b47d7da6b/logger.go#L374
So it will log debug, info, warning and error logs except trace level logs.

* `--logging_format={plaintext|json|CEF}`
  Logging format.

    * `plaintext` — (default) pretty human readable key/value format<br>
      ```
      time="2021-07-12T14:02:12+03:00" level=info msg="Starting service acra-translator [pid=475995]" version=0.85.0
      ```

    * `json` — one JSON object per line, easy to parse by most log collectors<br>
      ```
      {"level":"info","msg":"Starting service acra-translator [pid=476077]","product":"acra-translator","timestamp":"2021-07-12T14:02:50+03:00","unixTime":"1626087770.004","version":"0.85.0"}
      ```

    * `CEF` — Common Event Format<br>
      ```
      CEF:0|cossacklabs|acra-translator|0.85.0|100|Starting service acra-translator [pid\=476133]|1|unixTime=1626087782.510
      ```

## Web interface

### AcraServer settings

![](/files/acra-webconfig/acra-webconfig-ui.png)

> {{< hint info >}}
> After pressing `Save` button AcraServer dumps configuration to the configuration file and gracefully restarts applying all changes.
> {{< /hint >}}

Reviewer: ```suggestion
{{< hint info >}}
After pressing `Save` button AcraServer dumps configuration to the configuration file and gracefully restarts to apply all changes.
{{< /hint >}}
```

### AcraCensor & Zone settings

> Now `acra-webconfig` doesn't support changing AcraCensor's and Zone related parameters.

Reviewer: ```suggestion
Currently `acra-webconfig` does not support changing AcraCensor and Zone-related parameters.
```
